src/services/ipaPublicAdministrationService.ts

```diff
@@ -24,31 +27,37 @@ export function upsertFromIpa(): Promise<void> {
       indicepaResponse.body
         .pipe(csvStream)
         .pipe(
-          es.map((entry: IIpaPublicAdministration, cb: () => void) => {
-            if (entry.cf_validato !== "S") {
-              // filter out entries without a validated CF
-              return cb();
-            }
-            if (!entry.Cf || !entry.Cf.match(/^\d{2,}$/)) {
-              // filter out entries with bogus CF
-              return cb();
-            }
-            const entryIndex = notUpdatedEntries.findIndex(
-              entryFromDb => entryFromDb.cod_amm === entry.cod_amm
-            );
-            notUpdatedEntries.splice(entryIndex, 1);
-            IpaPublicAdministrationModel.upsert(entry)
-              .then(() => {
-                cb();
-              })
-              .catch(error => {
-                log.error(
-                  "IpaPublicAdministration upsert failed for %s: %s",
-                  entry.des_amm,
-                  error
+          es.map((parsedRow: IIpaPublicAdministrationRaw, cb: () => void) => {
+            IpaPublicAdministration.decode(
+              Object.entries(parsedRow)
+                .map(([key, val]) => [key, val.trim()])
+                .reduce(
+                  (prev, [cKey, cVal]) => {
+                    return { ...prev, [cKey]: cVal };
+                  },
+                  {} as unknown
+                )
+            ).fold(
+              () => cb(),
+              newEntry => {
+                const entryIndex = notUpdatedEntries.findIndex(
```

Comment:
consider using a `Set()` to check for duplicates. You may also chose the fp-ts implementation: 

https://github.com/gcanti/fp-ts/blob/1.12.0/src/Set.ts

Reply:
`notUpdatedEntries` can't have duplicate elements, because `cod_amm` is the primary key column of the `IpaPublicAdministrations` table.